File: nuxt_side/components/Swars/models/judge_info.js
import { ApplicationMemoryRecord } from "@/components/models/application_memory_record.js"

export class JudgeInfo extends ApplicationMemoryRecord {
  static get define() {
    return [
      { key: "win",  name: "勝ち",     },
      { key: "lose", name: "負け",     },
      { key: "draw", name: "引き分け", },
    ]
  }
}

File: app/models/quick_script/swars/cross_search_script/judge_info.rb
class QuickScript::Swars::CrossSearchScript
  class JudgeInfo
    include ApplicationMemoryRecord
    memory_record [
      { key: :win,  name: "勝ち",     },
      { key: :lose, name: "負け",     },
      { key: :draw, name: "引き分け", },
    ]
  end
end
